### PR TITLE
Send request test

### PR DESCRIPTION
**Pull request by cbranch101**
There were some branching paths to handle here, so I split out most of the functions into ternaries to make it so I didn't have to configure every aspect of the input to test each branch, and to make the stubbing process simpler

I don't think the splitting out into handlers takes away any readability from what was already very readable code, but I could be wrong about that

### CODE REVIEW

**Comment by cbranch101** [comment]
Any idea what's up with this source map error I'm getting on Travis? The phantomJS tests are running no problem on my local machine

**Comment by stephenlacy** [comment]
I'll look into the travis issue.
We usually have the test file name match the file of the functions been tested, is there any particular reason why you added `Test` to the end of the test files?

**Comment by cbranch101** [comment]
Code wise, none at all, mostly just so I don't get the test and the actual file confused when I have them both open in my editor, but I can change them back

**Comment by stephenlacy** [comment]
#12 will fix the tests - @contra will review/merge

**Comment by cbranch101** [comment]
@stevelacy Awesome! Thanks for looking into that.

File: src/lib/sendRequest.js [hunk 1]
```diff
@@ -26,32 +18,50 @@ export default (opt) => (dispatch) => {
...
-      err = new Error(`Unknown response type: '${res.type}' from ${debug}`)
-    }
-    if (err) {
-      if (opt.onError) opt.onError(err)
-      return dispatch({
-        type: 'tahoe.failure',
-        meta: opt,
-        payload: err
-      })
+export const handleRequest = (opt, dispatch, onTail, onStandard) => {
```
Comment (yocontra) [inline src/lib/sendRequest.js]: too many args, an obj would be clearer code

File: src/lib/sendRequest.js [hunk 2]
```diff
@@ -26,32 +18,50 @@ export default (opt) => (dispatch) => {
...
+    type: 'tahoe.failure',
+    meta: opt,
+    payload: err
   })
 }
+
+export const handleStandardRequest = (opt, dispatch, beforeEnd) => {
+  const req = request[opt.method.toLowerCase()](opt.endpoint)
+  beforeEnd(opt, req)
+  req.end((err, res) => handleResponse(err, res, opt, dispatch, handleError, handleSuccess))
```
Comment (yocontra) [inline src/lib/sendRequest.js]: too many args, an obj would be clearer code

**Comment by cbranch101** [comment]
@contra You were absolutely right about having too many arguments, I knew that was an issue before I pushed it, but I put off dealing with it because it didn't really effect the external api at all.  

How do you feel about splitting out into a params object and a handlers object going all the way down the branching tree?  The only thing that seems a little confusing is the difference between params and opt, but it still seems a lot cleaner. 

Let me know if this works.

File: src/lib/sendRequest.js [hunk 3]
```diff
@@ -26,32 +29,70 @@ export default (opt) => (dispatch) => {
   if (opt.withCredentials) {
     req.withCredentials()
   }
+}
 
-  req.end((err, res) => {
-    if (!res && !err) {
-      err = new Error(`Connection failed: ${debug}`)
+export const handleRequest = (params, { onTail, onStandard }) => {
```
Comment (yocontra) [inline src/lib/sendRequest.js]: `handleRequest({ options, dispatch, createRequest })` ?

File: src/lib/sendRequest.js [hunk 4]
```diff
@@ -26,32 +29,70 @@ export default (opt) => (dispatch) => {
...
+    payload: {
+      raw: res.body,
+      normalized: opt.model ? getEntities(res.body, opt) : null
     }
-    if (!err && res.type !== 'application/json') {
-      err = new Error(`Unknown response type: '${res.type}' from ${debug}`)
+  })
+}
+
+export const handleError = ({ err, opt, dispatch }) => {
```
Comment (yocontra) [inline src/lib/sendRequest.js]: keys in option objects should be descriptive (error / options)

File: src/lib/sendRequest.js (hunk 3 above)
Comment (cbranch101) [inline src/lib/sendRequest.js]: @contra The only issue I can see with this is that we have to repackage that params object every time we pass it down which happens quite a lot here, do you think that's better?  Or is there a way with destructuring to pull out variables and keep the wrapping object? I searched around but I could find any.

File: src/lib/sendRequest.js (hunk 4 above)
Comment (cbranch101) [inline src/lib/sendRequest.js]: Yah, I was cargo-culting a little bit here, I'll change that

**Comment by cbranch101** [comment]
I made the changes that you talked about, and I think they work for now, but I really don't like passing around all of these callbacks.  It makes setting up the tests so much more clunky, I'm going to mess around with some new ways to do this so that I can stub out functions without having to pass them in as arguments.

**Comment by yocontra** [comment]
@cbranch101 Yeah I think diving into the code and testing it in this way might not be the most useful. Really what we want to test is the actions and resulting changes to the store under different conditions, how things function internally doesn't really matter.

**Comment by yocontra** [comment]
I think with that in mind, I might drop the last commit for tests since it sacrificed a lot of code readability to test things that feel like overkill

**Pull request closed without merging**